api/tests/src/api/applications/test_application_routes.py

```diff
@@ -0,0 +1,100 @@
+import uuid
+
+import pytest
+from sqlalchemy import select
+
+from src.db.models.competition_models import Application, Competition
+from tests.src.db.models.factories import CompetitionFactory, OpportunityFactory
+
+
+@pytest.fixture(autouse=True)
+def clear_competitions(db_session):
+    db_session.query(Application).delete()
+    db_session.query(Competition).delete()
+    db_session.commit()
+    yield
```

Comment:
FYI - if there is no logic after the yield, it doesn't do anything (eg. the fixture that creates the test DB tables creates tables, yields, then deletes tables)